### ISSUE TITLE
read_Daybreak2R() may not properly recognise a binary file

### ISSUE THREAD

**Issue by mcol** [R-Lum/Luminescence#135]
If `read_Daybreak2R()` is called on a file with no `.DAT` extension, the following code is reached:

```R
file2read <- suppressWarnings(readLines(file))

##check whether this is a binary file
if(!all(charToRaw(file2read[1]) <= as.raw(127))){
  stop("[read_Daybreak2R()] The provided file is no ASCII-file and cannot be imported!", call. = FALSE)
}
```

However, this code is not enough to prevent an attempt to read a proper binary file (as opposed to a text file with non-ASCII characters). For example, this crashes:
```R
read_Daybreak2R(file = system.file("extdata/BINfile_V8.binx", package = "Luminescence"))
```
```console
[read_Daybreak] file extension not of type '.DAT' try to import ASCII-file ... 
Error in records.row_number[x]:length(file2read) : NA/NaN argument
```
A possible mitigation is to call `readLines(file, ok = FALSE)`, which raise an error directly when reading in the file if its length is not what is expected, which happens for the .binx file used above.

**Comment by RLumSK** [comment]
@mcol I do agree, please create a PR.

**Comment by mcol** [comment]
It turns out that using `readLines(file, ok = FALSE)` is too restrictive and would reject also good input files, but I've found a different approach that fixes the error and makes the existing tests pass.

**Comment by RLumSK** [comment]
Closed with PR#140, thanks!
